addon/adapters/osf-adapter.js

```diff
@@ -68,21 +37,64 @@ export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
         var links = relationship ? snapshot.record.get(
             `relationshipLinks.${Ember.String.underscore(relationship)}.links`
         ) : false;
-        if (links && links.self || links.related) {
+        if (links && (links.self || links.related)) {
             return links.self ? links.self.href : links.related.href;
         } else {
             return this.buildURL(...arguments);
         }
     },
+    _serializeHasMany(serialized) {
+        if (serialized.length > 1) {
+            serialized = {
+                data: serialized.map(function(record) {
+                    var data = record.data;
+                    return data;
+                })
+            };
+        } else {
+            serialized = serialized[0];
+        }
+        return serialized;
+    },
+    _handleManyRequest(store, type, snapshot, query, relationship, serializer) {
+        var relationMeta = snapshot.record[relationship].meta();
+        var serialized = snapshot.hasMany(relationship).filter(each => each.id === null || Object.keys(each.changedAttributes()).length).map(each => serializer.serialize(each));
+        var options = Ember.$.extend({},
+            {
+                requestType: () => 'PATCH',
```

Comment:
to allow for relationships that use more than one requestType, this needs to be called.